### ISSUE TITLE
Redirect URL throws http error

### ISSUE THREAD

**Issue by suguna-e** [manjitlatthe/ionic-php-ccavenue-integration#5]
Hi,

Thanks for the code you have provided. On completing transactions, the ccavenue checkout page is not redirecting to the ionic app and it throws http error on hitting ccavmobileresponsehandler page. Please help us to get success or failure messages on completing the transactions.

**Comment by manjitlatthe** [comment]
Enable API support for your CCavenue merchant account by Whitelist your outgoing IP address xxx.xx.xx.xxx for API calls. Send the request regarding this to service@ccavenue.com with your valid Merchant ID.

Are you using inappbrowser?

**Comment by suguna-e** [comment]
We have done all that and we are happy that the transaction is happening. Only error we are encountering is that http error 500, may be that we didn't mention any redirect url i guess. Yes, we are using inappbrowser and we are trying to get back the values from inappbrowser to app on completing transactions.

**Comment by manjitlatthe** [comment]
The default response URL is set to $responseURL = $base_url.'backend/ccavMobileResponseHandler.php';
in getRSA.php file. So you don't need to set by yourself.

Use localStorage to get the values from Inappbrowser to Ionicapp.

**Comment by suguna-e** [comment]
can you please provide a sample code for that as we are new to angularjs and payment gateway integration

**Comment by suguna-e** [comment]
can you please provide a sample code for that as we are new to angularjs and payment gateway integration

**Comment by suguna-e** [comment]
I can get the values from browser to app through localstorage. But why I am not getting transaction successful message in the browser itself?

**Comment by manjitlatthe** [comment]
Please use the updated ccavMobileResponseHandler.php file.

**Comment by suguna-e** [comment]
We have used that updated file and we are still encountering that error.

**Comment by manjitlatthe** [comment]
Can I check your live code on server?

**Comment by suguna-e** [comment]
I need to get approval from my superior to give server access. Could you please support us on Monday (11-09-2017).

**Comment by manjitlatthe** [comment]
As per availability if possible I will.

**Comment by suguna-e** [comment]
Can I share my teamviewer ID?

**Comment by suguna-e** [comment]
Can I share my teamviewer ID?

**Comment by manjitlatthe** [comment]
yes

**Comment by suguna-e** [comment]
ID: 310-324-265
Pwd: 6016

**Comment by manjitlatthe** [comment]
If you get latest Crypto.php file please let me know.

**Comment by suguna-e** [comment]
Sure.Thank You Very Much!!

**Comment by suguna-e** [comment]
I have updated the Crypto file but still we are encountering the same error. On checking the same, we found that the Crypto file of ccavenue is no different from your Crypto file.

**Comment by suguna-e** [comment]
Do we need to upload ccavRequestHandler.php file in the path of the backend folder?

**Comment by suguna-e** [comment]
I have updated the Crypto file but still we are encountering the same error. On checking the same, we found that the Crypto file of ccavenue is no different from your Crypto file.

**Comment by manjitlatthe** [comment]
Have you asked ccavenue support about that issue? Because the issue is with the decrypt method which is placed in Crypto.php file.

Also replace "error_reporting(0);" to "error_reporting(E_ALL);" and then check.

**Comment by manjitlatthe** [comment]
Is your problem solved?

**Comment by suguna-e** [comment]
No..We have tried changing that too, but it does not help us..We have raised a query with ccavenue, they are saying that there is an issue with the internal server. We are hopeless at this juncture.

**Comment by manjitlatthe** [comment]
Did you told them that, you are getting this error on calling of "decrypt" function?
There is nothing else we did in our code.

**Comment by suguna-e** [comment]
I have followed up with the CCAvenue.  They are asking us to check the code

**Comment by suguna-e** [comment]
I have followed up with the CCAvenue. They are asking us to check the code

**Comment by suguna-e** [comment]
I have followed up with the CCAvenue. They are asking us to check the code
